fondant/dataset.py

```diff
@@ -95,71 +114,145 @@ def load_dataframe(self, spec: FondantComponentSpec) -> dd.DataFrame:
 
         return df
 
-    def _upload_index(self, df: dd.DataFrame):
-        # get remote path
+    @staticmethod
+    def _create_write_dataframe_task(
+        *, df: dd.DataFrame, remote_path: str, schema: t.Dict[str, str]
+    ) -> dd.core.Scalar:
+        """
+        Creates a delayed Dask task to upload the given DataFrame to the remote storage location
+         specified in the manifest.
+        Args:
+            df: The DataFrame to be uploaded.
+            remote_path: the location to upload the subset to
+            schema: the schema of the dataframe to write
+        Returns:
+             A delayed Dask task that uploads the DataFrame to the remote storage location when
+              executed.
+        """
+
+        # Define task to upload index to remote storage
+        upload_index_task = dd.to_parquet(
+            df, remote_path, schema=schema, overwrite=True, compute=False
+        )
+        return upload_index_task
+
+    def get_upload_index_task(self, df: dd.DataFrame) -> dd.core.Scalar:
+        """
+        Create a Dask task that uploads the index dataframe to a remote location.
+        Args:
+            df: The input Dask dataframe.
+        Returns:
+            A Dask scalar representing the upload task.
+        """
         remote_path = self.manifest.index.location
+        index_columns = list(self.manifest.index.fields.keys())
 
-        # upload to the cloud
-        dd.to_parquet(
-            df,
-            remote_path,
-            schema=self.index_schema,
-            overwrite=True,
+        # load index dataframe
+        index_df = df[index_columns]
+
+        upload_index_task = self._create_write_dataframe_task(
+            df=index_df, remote_path=remote_path, schema=self.index_schema
         )
 
-    def _upload_subset(
-        self, name: str, fields: t.Mapping[str, Field], df: dd.DataFrame
-    ):
-        # add subset to the manifest
-        manifest_fields = [
-            (field.name, Type[field.type.name]) for field in fields.values()
-        ]
-        self.manifest.add_subset(name, fields=manifest_fields)
+        return upload_index_task
 
-        # create expected schema
-        expected_schema = {field.name: field.type.name for field in fields.values()}
-        expected_schema.update(self.index_schema)
+    def get_upload_subsets_task(
```

Comment:
```suggestion
    def get_upload_subsets_tasks(
```
Seems like this method returns a list of tasks